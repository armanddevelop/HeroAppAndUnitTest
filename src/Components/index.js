export { HeroList } from "./Hero/HeroList";

export { Hero } from "./Hero/Hero";

export { Navbar } from "./UI/NavBar";

export { Login } from "./Login/Login";

export { SearchScreen } from "./Search/SearchScreen";

export { DcScreen } from "./DC/DcScreen";

export { MarvelScreen } from "./Marvel/MarvelScreen";
