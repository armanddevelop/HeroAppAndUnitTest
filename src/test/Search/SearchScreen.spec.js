import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import { SearchScreen } from "../../Components";

const mockNavigate = jest.fn();
const onSubmit = jest.fn();
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockNavigate,
}));
const e = {
  target: {
    value: "batman",
    name: "searchText",
  },
  preventDefault: () => {},
};
describe("Test to SearchScreen component", () => {
  it("Snapshot to SearchScreen component", () => {
    const container = mount(
      <MemoryRouter initialEntries={["/search"]}>
        <SearchScreen />
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
    expect(container.find(".alert-info").text().trim()).toBe("Busca un heroe");
  });
  it("Should render a hero with value of query string", () => {
    const q = "Batman";
    const container = mount(
      <MemoryRouter initialEntries={[`/search?q=${q}`]}>
        <SearchScreen />
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
    expect(container.find("h5").text().trim()).toBe(q);
    expect(container.find("input").prop("value")).toBe(q);
  });
  it("Should render error if no coincidences", () => {
    const q = "licha";
    const container = mount(
      <MemoryRouter initialEntries={[`/search?q=${q}`]}>
        <SearchScreen />
      </MemoryRouter>
    );

    expect(container.find(".alert-danger").exists()).toBeTruthy();
  });
  it("Should change the navigate to the new url", () => {
    const container = mount(
      <MemoryRouter initialEntries={[`/search`]}>
        <SearchScreen />
      </MemoryRouter>
    );
    container.find("input").simulate("change", e);
    container.find("form").prop("onSubmit")(e);

    expect(mockNavigate).toHaveBeenCalledWith(`?q=${e.target.value}`);
  });

  it("Should button search submit form", () => {
    const container = mount(
      <MemoryRouter initialEntries={[`/search`]}>
        <SearchScreen />
      </MemoryRouter>
    );
    container.find("button").prop("type", () => {
      onSubmit(e);
    });

    expect(onSubmit).not.toHaveBeenCalled();
  });
});
