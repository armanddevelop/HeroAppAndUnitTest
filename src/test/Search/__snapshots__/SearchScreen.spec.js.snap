// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to SearchScreen component Should render a hero with value of query string 1`] = `
Array [
  <h1>
    Search Screen
  </h1>,
  <hr />,
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Search Form
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Serach by Hero name"
          type="text"
          value="Batman"
        />
        <button
          className="btn btn-outline-primary mt-1"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Resultados
      </h4>
      <hr />
      <div
        className="col animate__animated animate__fadeIn"
      >
        <div
          className="card"
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-4"
            >
              <img
                alt="Batman"
                className="card-img"
                src="/assets/heroes/dc-batman.jpg"
              />
            </div>
            <div
              className="col-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Batman
                </h5>
                <p
                  className="card-text"
                >
                  Bruce Wayne
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    Detective Comics #27
                  </small>
                </p>
                <a
                  href="/hero/dc-batman"
                  onClick={[Function]}
                >
                  Mas Info
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Test to SearchScreen component Snapshot to SearchScreen component 1`] = `
Array [
  <h1>
    Search Screen
  </h1>,
  <hr />,
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Search Form
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Serach by Hero name"
          type="text"
          value=""
        />
        <button
          className="btn btn-outline-primary mt-1"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Resultados
      </h4>
      <hr />
      <div
        className="alert alert-info"
      >
        Busca un heroe
      </div>
    </div>
  </div>,
]
`;
